src/jit/importer.cpp

```diff
@@ -7522,7 +7522,9 @@ var_types Compiler::impImportCall(OPCODE                  opcode,
             }
         }
 
-        if (!bIntrinsicImported)
+        // Don't create cast for a variable on the stack if it is intrinsic or tail call.
+        // We assume that the value will not be physically represented there.
+        if (!bIntrinsicImported && !(tailCall && canTailCall))
```

Comment:
I am inclined to believe that var=cast(call) is a new tail call pattern that is possible which is not currently handled by morph logic.  But what stumps me that so far we have not encountered such a tail call pattern to expose this issue.  Do we have a test case that repro's this issue in non-R2R context?  We need it anyways for adding as a unit test case.

I think option #2 is the way to go after we have convinced ourselves that this can happen in non-R2R tail call scenarios by constructing a repro case.